### ISSUE TITLE
Currency symbol abbreviation "Sc" is not a valid property

### ISSUE THREAD

**Issue by aikow** [rust-lang/regex#835]
#### What version of regex are you using?

1.5.4

#### Describe the bug at a high level.

The unicode abbreviation for for the currency symbols as defined here https://www.unicode.org/Public/UCD/latest/ucd/PropertyValueAliases.txt
is not valid and gives a regex parse error.

#### What are the steps to reproduce the behavior?

Add the regex crate to the cargo.toml file

File: `Cargo.toml`
```toml
[dependencies]
regex = "1.5.4"
```

File: `main.rs`
```rust
use regex::Regex;

fn main() {
    let regex = Regex::new(r"\p{Sc}").unwrap();
    if regex.is_match("$") {
        println!("Hello, world!");
    }
}
```

#### What is the actual behavior?

```sh
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Syntax(
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
regex parse error:
    \p{Sc}
    ^^^^^^
error: Unicode property not found
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
)', src/main.rs:4:39
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

#### What is the expected behavior?

Expected the regex to compile and return the "$" as a match.

**Comment by BurntSushi** [comment]
TL;DR - It's a bug and you can work around it for now by using `\p{gc=Sc}` (or `\p{gc:Sc}`) to force the class parser to treat `Sc` as a general category. Although, personally, I would use `\p{CurrencySymbol}`, since I think that's clearer.

OK, so `Sc` is an abbreviation for the _general category_ called `Currency`, and that's defined by [Unicode in UTS#18](https://unicode.org/reports/tr18/#General_Category_Property). So first question is, do we have that general category and abbreviation in our Unicode tables? [Yes](https://github.com/rust-lang/regex/blob/5197f21287344d2994f9cf06758a3ea30f5a26c3/regex-syntax/src/unicode_tables/general_category.rs#L271) and [yes](https://github.com/rust-lang/regex/blob/5197f21287344d2994f9cf06758a3ea30f5a26c3/regex-syntax/src/unicode_tables/property_values.rs#L133).

The next thing I noticed is that `sc` is actually a way to explicitly specify "script extension." So, e.g., `\p{Greek}` and `\p{sc=Greek}` are equivalent. So my thinking now is that the special treatment of `sc` is messing with the general category lookup. I believe [this](https://github.com/rust-lang/regex/blob/5197f21287344d2994f9cf06758a3ea30f5a26c3/regex-syntax/src/unicode.rs#L191) is where cases like `\p{xxx}` are handled, which to me implies that something is going wrote in the [`canonical_binary`](https://github.com/rust-lang/regex/blob/5197f21287344d2994f9cf06758a3ea30f5a26c3/regex-syntax/src/unicode.rs#L237-L258) method.

... but that all looks right to me. And I don't see where any special handling of `sc` might be getting in the way. But this is just all by inspection. I write this down in case someone else wants to take a look at fixing this.
